qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -232,26 +238,67 @@ def _metadata(self):
 
         Returns:
             A list of transpiled circuits.
         """
-        transpile_opts = copy.copy(self.transpile_options.__dict__)
-        if "inst_map" in transpile_opts:
-            LOG.warning(
-                "Instruction schedule maps should not be present in calibration "
-                "experiments. Overriding with the inst. map of the calibrations."
-            )
+        transpiled = []
+        for circ in self.circuits():
+            circ = self._map_to_physical_qubits(circ)
+            self._attach_calibrations(circ)
+
+            if self.experiment_options.add_measure_schedules:
+                self._attach_measure_schedules(circ)
+
+            transpiled.append(circ)
+
+        return transpiled
 
-        transpile_opts["inst_map"] = self.calibrations.default_inst_map
-        transpile_opts["initial_layout"] = list(self.physical_qubits)
+    def _map_to_physical_qubits(self, circuit: QuantumCircuit) -> QuantumCircuit:
+        """Map program qubits to physical qubits.
 
-        return transpile(self.circuits(), self.backend, **transpile_opts)
+        Args:
+            circuit: The quantum circuit to map to device qubits.
+
+        Returns:
+            A quantum circuit that has the same number of qubits as the backend and where
+            the physical qubits of the experiment have been properly mapped.
+        """
+        initial_layout = Layout.from_intlist(list(self.physical_qubits), *circuit.qregs)
+
+        return PassManager(
+            [
+                SetLayout(initial_layout),
+                FullAncillaAllocation(CouplingMap(self._backend_data.coupling_map)),
+                EnlargeWithAncilla(),
+                ApplyLayout(),
+            ]
+        ).run(circuit)
+
+    def _attach_measure_schedules(self, circuit: QuantumCircuit):
+        """Attach measurement schedules to the circuit."""
+        for qubit in self.physical_qubits:
+            meas_sched = self._cals.get_schedule("measure", qubit)
+            circuit.add_calibration("measure", (qubit,), meas_sched)
+
+    @abstractmethod
+    def _attach_calibrations(self, circuit: QuantumCircuit):
+        """Attach the calibrations to the quantum circuit.
+
+        This method attaches calibrations from the `self._cals` instance to the transpiled
+        quantum circuits. Given how important this method is it is made abstract to force
+        potential calibration experiment developers to implement it and think about how
+        schedules are attached to the circuits. The implementation of this method is delegated
+        to the sub-classes so that they can map gate instructions to the schedules stored in the
+        ``Calibrations`` instance. This method is needed for most calibration experiments. However,
+        some experiments already attach circuits to the logical circuits and do not needed to run
+        ``_attach_calibrations``. In such experiments a simple ``pass`` statement will suffice.
+        """
```

Comment:
```suggestion
    def _attach_calibrations(self, circuit: QuantumCircuit):
        """Attach the calibrations to the quantum circuit.

        This method attaches calibrations from the `self._cals` instance to the transpiled
        quantum circuits. Given how important this method is it is made abstract to force
        potential calibration experiment developers to implement it and think about how
        schedules are attached to the circuits. The implementation of this method is delegated
        to the sub-classes so that they can map gate instructions to the schedules stored in the
        ``Calibrations`` instance. This method is needed for most calibration experiments. However,
        some experiments already attach circuits to the logical circuits and do not needed to run
        ``_attach_calibrations``. In such experiments a simple ``pass`` statement will suffice.
        """
        pass
```
As you wrote in the Rabi some experiments may not require calibration instance. Abstract method is one that all subclasses must define. You can do nothing by default, i.e. just `pass`

Reply:
Indeed. I purposefully avoided making this a default method with just `pass`. The reason for this is that I want the developer to consciously think about how the calibrations are attached. If we go with `pass` and make this a default method then the developer might forget to implement this which could result in undesired behavior which is hard to detect (e.g. the backend has all the default gates and the cals are not provided to override them).

Reply:
Fair enough.